Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+""" The RucioFileCatalogClient is a class that allows to interface Dirac with Rucio
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+__RCSID__ = "$Id"
+
+import os
+import uuid
+import datetime
+from copy import deepcopy
+
+import DIRAC
+from DIRAC                                              import S_OK, S_ERROR, gLogger
+from DIRAC.Resources.Catalog.Utilities                  import checkCatalogArguments
+from DIRAC.Core.Utilities.List                          import breakListIntoChunks
+from DIRAC.Core.Security.ProxyInfo                      import getProxyInfo
+from DIRAC.Resources.Catalog.FileCatalogClientBase import FileCatalogClientBase
+
+from rucio.client import Client
+from rucio.common.exception import DataIdentifierNotFound, FileReplicaAlreadyExists, \
+                                   CannotAuthenticate, MissingClientParameter
+from rucio.common.utils import chunks, extract_scope
+sLog = gLogger.getSubLogger(__name__)
+
+
+def get_scope(lfn, scopes=[]):
+  """Helper function that extract the scope from the LFN
+  """
+  scope, _ = extract_scope(did=lfn, scopes=scopes)
+  return scope
+
+
+class RucioFileCatalogClient(FileCatalogClientBase):
+
+  READ_METHODS = FileCatalogClientBase.READ_METHODS + ['isLink', 'readLink', 'isFile', 'getFileMetadata',
+                                                       'getReplicas', 'getReplicaStatus', 'getFileSize',
+                                                       'isDirectory', 'getDirectoryReplicas',
+                                                       'listDirectory', 'getDirectoryMetadata',
+                                                       'getDirectorySize', 'getDirectoryContents',
+                                                       'resolveDataset', 'getLFNForPFN', 'getUserDirectory']
+
+
+  WRITE_METHODS = FileCatalogClientBase.WRITE_METHODS + ['createLink', 'removeLink', 'addFile', 'addReplica',
+                                                         'removeReplica', 'removeFile', 'setReplicaStatus',
+                                                         'setReplicaHost', 'createDirectory', 'removeDirectory',
+                                                         'removeDataset', 'removeFileFromDataset', 'createDataset',
+                                                         'changePathOwner', 'changePathMode']
+
+
+  NO_LFN_METHODS = FileCatalogClientBase.NO_LFN_METHODS + ['getUserDirectory', 'createUserDirectory',
+                                                           'createUserMapping', 'removeUserDirectory']
+
+
+  ADMIN_METHODS = FileCatalogClientBase.ADMIN_METHODS + ['getUserDirectory', 'createUserDirectory',
+                                                         'createUserMapping', 'removeUserDirectory']
+
+  def __init__(self, **options):
+    """ Constructor
+    """
+    self.convertUnicode = True
```

Comment:
Can you explain what this is trying to achieve? In particular is `str(...)` the right thing to do for Python 3 as well? Or should it be `.encode()`?